src/runtime_src/xdp/profile/device/aie_trace/ve2/aie_trace_offload_ve2.cpp

```diff
@@ -0,0 +1,572 @@
+/**
+ * Copyright (C) 2019-2022 Xilinx, Inc
+ * Copyright (C) 2022-2024 Advanced Micro Devices, Inc. - All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#define XDP_PLUGIN_SOURCE
+
+#include <iostream>
+
+#include "core/common/message.h"
+#include "core/include/xrt/xrt_kernel.h"
+#include "xdp/profile/database/database.h"
+#include "xdp/profile/database/static_info/aie_constructs.h"
+#include "xdp/profile/device/aie_trace/ve2/aie_trace_logger_ve2.h"
+#include "xdp/profile/device/aie_trace/ve2/aie_trace_offload_ve2.h"
+#include "xdp/profile/device/pl_device_intf.h"
+#include "xdp/profile/plugin/aie_trace/x86/aie_trace_kernel_config.h"
+#include <unistd.h>
+
+/*
+ * XRT_X86_BUILD is set only for x86 builds
+ * Only compile this on edge+versal build
+ */
+#if defined (XRT_ENABLE_AIE) && ! defined (XRT_X86_BUILD) && ! defined (XDP_CLIENT_BUILD)
```

Comment:
Are these macro guards required for VE2 flow? The current cpp file is compiled only under XDP_VE2_BUILD_CMAKE. Can XRT_X86_BUILD, XDP_CLIENT_BUILD be ever set for VE2? Also, can AIE be disabled for VE2?